app/uk/gov/hmrc/apiplatformjobs/scheduled/SendUnusedApplicationNotificationsJob.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apiplatformjobs.scheduled
+
+import javax.inject.{Inject, Singleton}
+import org.joda.time.DateTime
+import play.api.Configuration
+import play.modules.reactivemongo.ReactiveMongoComponent
+import uk.gov.hmrc.apiplatformjobs.connectors.EmailConnector
+import uk.gov.hmrc.apiplatformjobs.models.Environment.{Environment, PRODUCTION, SANDBOX}
+import uk.gov.hmrc.apiplatformjobs.repository.UnusedApplicationsRepository
+
+import scala.concurrent.{ExecutionContext, Future}
+
+abstract class SendUnusedApplicationNotificationsJob(unusedApplicationsRepository: UnusedApplicationsRepository,
+                                                     emailConnector: EmailConnector,
+                                                     environment: Environment,
+                                                     configuration: Configuration,
+                                                     mongo: ReactiveMongoComponent)
+  extends UnusedApplicationsJob("SendUnusedApplicationNotificationsJob", environment, configuration, mongo) {
+
+  override def functionToExecute()(implicit executionContext: ExecutionContext): Future[RunningOfJobSuccessful] = {
+    val notificationTime = DateTime.now
+
+    for {
+      appsToNotify <- unusedApplicationsRepository.unusedApplicationsToBeNotified(environment, notificationTime)
```

Comment:
There is a lot going on here and in email connector.
I would break this down further into functions so its easier to follow.

Reply:
That I can do...